lib/maths/CBoostedTreeFactory.cc

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#include <maths/CBoostedTreeFactory.h>
+
+#include <maths/CBayesianOptimisation.h>
+#include <maths/CBoostedTreeImpl.h>
+
+namespace ml {
+namespace maths {
+
+CBoostedTreeFactory::TBoostedTreeUPtr CBoostedTreeFactory::build() {
+    this->initializeMissingFeatureMasks(*m_Frame);
+    std::tie(m_Tree->m_Impl->m_TrainingRowMasks,
```

Comment:
I find the `m_Tree->m_Impl->` distracting reading the code in this class. How about wrapping this up in a private function `CBoostedTreeFactory::tree()` which return the implementation by writable reference?